rust/gelato/src/task_status_call.rs

```diff
@@ -0,0 +1,121 @@
+pub struct TaskStatusCallResult {
+    pub data: Vec<TransactionStatus>,
+}
+
+#[derive(Debug)]
+pub struct TaskStatusCall {
+    pub http: Arc<reqwest::Client>,
+    pub args: TaskStatusCallArgs,
+}
+impl TaskStatusCall {
+    #[instrument]
+    pub async fn run(&self) -> Result<TaskStatusCallResult, GelatoError> {
+        let url = format!(
+            "{}/tasks/GelatoMetaBox/{}",
+            RELAY_URL, self.args.task_id
+        );
+        info!(?url);
+        let res = self.http.get(url).send().await?;
+        info!(?res);
+        let result = TaskStatusCallResult::from(res.json().await?);
+        Ok(TaskStatusCallResult::from(result))
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
+pub enum TaskStatus {
+    CheckPending,
+    ExecPending,
+    ExecSuccess,
+    ExecReverted,
+    WaitingForConfirmation,
+    Blacklisted,
+    Cancelled,
+    NotFound,
+}
+
+#[derive(Debug, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TransactionStatus {
+    pub service: String,
+    pub chain: String,
+    pub task_id: String,
+    pub task_state: TaskStatus,
+    pub created_at: Option<String>,
+    pub last_check: Option<Check>,
+    pub execution: Option<Execution>,
+    pub last_execution: Option<String>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Execution {
+    pub status: String,
+    pub transaction_hash: String,
+    pub block_number: u64,
+    pub created_at: Option<String>,
+}
+
+// Sometimes the value corresponding to the 'last_check'
+// key is a string timestamp, other times it is filled in
+// with lots of detailed fields. Represent that with
+// an enum and let serde figure it out.
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(untagged)]
+pub enum Check {
+    Timestamp(String),
+    CheckWithMetadata(CheckInfo),
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CheckInfo {
+    pub task_state: TaskStatus,
+    pub message: String,
+    pub payload: Payload,
+    pub chain: Option<String>,
+    pub created_at: Option<String>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Payload {
+    pub to: String,
+    pub data: String,
+    pub type_: String,
+    pub fee_data: FeeData,
+    pub fee_token: String,
+    pub gas_limit: BigNumType,
+    pub is_flashbots: Option<bool>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FeeData {
+    pub gas_price: BigNumType,
+    pub max_fee_per_gas: BigNumType,
+    pub max_priority_fee_per_gas: BigNumType,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BigNumType {
+    pub hex: String,
+    pub type_: String,
```

Comment:
Uh, does this work as expected with the `camelCase` setting? I could see it stripping the `_`

Reply:
It does work. camelCase indeed strips the '_' and that is the intended effect. I can't name the field 'type' because that is a Rust keyword, but the json handed to us in the http response names the field 'type'. Maybe it'd be more idiomatic if I named it something like 'type_field' and did a #[serde_rename] ?

Reply:
I've also now discovered rust's escaping capability wherein I could maybe do something like:

...
pub struct BigNumType {
    pub hex: String,
    pub r#type: String,
}

Reply:
Yeah, this is one of the only times where I actually can see the ident escaping making sense. So I guess I leave this up to you which option you want to use.